pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -233,10 +231,15 @@ func checkForError(getError error, existingES *esv1beta1.ExternalSecret) string
 }
 
 func getRemovedNamespaces(nsList v1.NamespaceList, provisionedNs []string) []string {
-	result := []string{}
+	var result []string
+
+	nsSet := map[string]struct{}{}
+	for i := range nsList.Items {
+		nsSet[nsList.Items[i].Name] = struct{}{}
+	}
```

Comment:
It is not directly related to the PR, but it just caught my eye, so I fixed it. The double for loop can be quite ineffective if the `nsList.Items` is long. I believe it is more effective if we use simply use a set.